hw/ip/pattgen/doc/_index.md

```diff
@@ -0,0 +1,143 @@
+This module conforms to the [Comportable guideline for peripheral functionality.]({{< relref "doc/rm/comportability_specification/index.md" >}})
+See that document for integration overview within the broader top-level system.
+
+## Features
+
+- Generates time-dependent patterns on two (2) channels, each with its own clock.
+-    In each channel, data is transmitted serially on a one-bit data (pda) output, which is synchronous to a corresponding parallel clock signal (pcl).
+   -    The channels may operate independently or synchronously with each other. 
+- Each output channel supports the following configuration registers:
+    -	Pattern data per output (2-64 bits of data).
+    -	Pattern length per output (2-64 bits).        
+    -	32-bit pre-divider to derive pattern clock from I/O clock (minimum value: 2).
+    -	A 10-bit repeat counter
+    -   A bit control the polarity of the clock signal 
+- Sends interrupt on pattern completion.
+    
+## Description 
+
+The PATTGEN IP transmits short (maximum 64 bits) time-dependent data patterns on two clock-parallel  channels.
+Each Channel consists of one clock (`pcl`) and one data (`pda`) line.
+The output channels may be activated and operated independently, or they can be started at the same time to effectively create a 4-output pattern.
+ 
+## Compatibility
+
+This IP block does not have any direct hardware compatibility requirements.
+
+# Theory of Operations 
+
+To start pattern generation, the registers interface of PATTGEN HWIP should be properly initialized and configured.
+To configure a single channel:
+1. Before configuration, disable the desired channel by clearing the corresponding enable bit: ({{< "regref PATT_CTRL.ENABLE_CH0" >}} for Channel 0, {{< regref "PATT_CTRL.ENABLE_CH1" >}} for Channel 1).
+1. Set the polarity bit ({{< regref "PATT_CTRL.POLARITY_CH0" >}} for Channel 0, {{< regref "PATT_CTRL.POLARITY_CH1" >}} for Channel 1) to determine the desired clock phase.
+For either channel, a zero in the polarity bit indicates that the channel clock line (`pcl`) should start low, and the channel data line `pda` transition on every falling edge of `pcl`.
+A one in the polarity bit inverts the `pcl` clock so that it starts high and `pda` transitions on the rising edge.
+{{<wavejson>}}
+{signal: [
+  {name: 'PATT_CTRL.ENABLE_CH0', wave: 'lh......'},
+  {name: 'PATT_CTRL.POLARITY_CH0 (default: low)', wave: '0.......'},
+  {name: 'pcl0_tx', wave: '0.hlhlhl'},
+  {name: 'pda0_tx', wave: 'x3.3.3.3', data: 'DATA[0] DATA[1] DATA[2]'},
+  {name: 'PATT_CTRL.POLARITY_CH1 (high)', wave: '1.......'},
+  {name: 'pcl1_tx', wave: '1.lhlhlh'},
+  {name: 'pda1_tx', wave: 'x5.5.5.5', data: 'DATA[0] DATA[1] DATA[2]'},
+  
+],
+  head: {text: 'Effect of the Polarity Registers',tick:0}}
+{{</wavejson>}}
+
+1. Program the seed patterns (1-64 bits in length) using the multi-registers {{< regref "PATT_DATA_CH0" >}} (for Channel 0) or {{<regref "PATT_CH1_DATA" >}} (for Channel 1).
```

Comment:
```suggestion
1. Program the seed patterns (between 1 and 64 bits in length) using the multi-registers {{< regref "PATT_DATA_CH0" >}} (for Channel 0) or {{<regref "PATT_CH1_DATA" >}} (for Channel 1).
```

I might be missing something, but sometimes it's between 1 and 64 bits, and sometimes it's between 2 and 64. Is this intentional?